lib/ansible/plugins/action/net_template.py

```diff
@@ -93,6 +94,11 @@ def _handle_template(self):
         except IOError:
             return dict(failed=True, msg='unable to load src file')
 
+        searchpath = [working_path, os.path.dirname(source)]
+        if self._task._role is not None:
+            searchpath.insert(1, C.DEFAULT_ROLES_PATH)
```

Comment:
Thank you @bcoca for your quick reply!  I've actually followed the existing template.py as a sample.  Let me know if the issue above also exists in template.py or net_template.py is another beast which we need to play differently.

https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/action/template.py#L125

Thank you for your support!

Cheers,